### PR TITLE
[release/v1.28.x] Cache maven local

### PR DESCRIPTION
**Pull request by bryan-aguilar**
Remake of #488 but push to main repo instead of fork.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/main-build.yml [hunk 1]
```diff
@@ -61,6 +61,14 @@ jobs:
           java-version: 17
           distribution: temurin
 
+      - name: Cache local Maven repository
```
Comment (rapphil) [inline .github/workflows/main-build.yml:64]: this needs to go after we publish everything to maven local no? in line 72

Comment (bryan-aguilar) [reply .github/workflows/main-build.yml:64]: You insert the action at the point in which you want to restore the cache. GitHub actions will automatically cache the path after job success.

File: .github/workflows/main-build.yml [hunk 2]
```diff
@@ -61,6 +61,14 @@ jobs:
...
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
```
Comment (rapphil) [inline .github/workflows/main-build.yml:68]: we don't use `pom.xml`. shouldn't this be `.kts`?

Comment (bryan-aguilar) [reply .github/workflows/main-build.yml]: Took another go at the key hash. Opted for a much more targeted hash that included the patch file hash.

**Review by rapphil — APPROVED** [review]

**Pull request merged**